cmdstanpy/cmdstan_args.py

```diff
@@ -381,9 +381,12 @@ def validate(self) -> None:
                             ' found {}'.format(self.seed[i])
                         )
 
-        if self.data is not None:
+        if isinstance(self.data, str):
             if not os.path.exists(self.data):
                 raise ValueError('no such file {}'.format(self.data))
+        elif not isinstance(self.data, dict):
+            if isinstance(self.method_args, OptimizeArgs):
```

Comment:
include SamplerArgs as well here please

Reply:
@mitzimorris I see in many `SamplerArgsTest` tests that data is null - is that expected to run sampling without any data file?

Reply:
yes, models without a data block are valid.

one use case is modelswhere the transformed data block is used to generate data for each chain/run via calls to the RNG functions.

Reply:
I've fixed the logic, now we throw exception when data is None *and* we are optimizing, or when data is different than str or dict.